function_wrapper_test.go

```diff
@@ -0,0 +1,56 @@
+package gosnowflake
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"testing"
+)
+
+var (
+	goWrapperCalled chan struct{}
+	once       sync.Once
+)
+
+// Initialize the goWrapperCalled channel which we will use for testing the goroutine wrap function
+func initGoWrapperCalledChannel() {
+	once.Do(func() {
```

Comment:
All this is a bit complex - maybe we can just remember some global variable when test wrapper is invoked instead of using and closing channels?

Reply:
Ok, will do